src/yunohost/app.py

```diff
@@ -1057,11 +1081,24 @@ def app_remove(operation_logger, app):
     operation_logger.extra.update({'env': env_dict})
     operation_logger.flush()
 
-    if hook_exec('/tmp/yunohost_remove/scripts/remove', args=args_list,
-                 env=env_dict)[0] == 0:
-        logger.success(m18n.n('app_removed', app=app))
+    try:
+        ret = hook_exec('/tmp/yunohost_remove/scripts/remove',
+                        args=args_list,
+                        env=env_dict)[0]
+    # Here again, calling hook_exec could failed miserably, or get
+    # manually interrupted (by mistake or because script was stuck)
+    # In that case we still want to proceed with the rest of the
+    # removal (permissions, /etc/yunohost/apps/{app} ...)
+    except (KeyboardInterrupt, EOFError, Exception):
+        ret = -1
+        import traceback
+        logger.exception(m18n.n('unexpected_error', error=u"\n" + traceback.format_exc()))
 
+    if ret == 0:
+        logger.success(m18n.n('app_removed', app=app))
         hook_callback('post_app_remove', args=args_list, env=env_dict)
+    else:
+        logger.warning(m18n.n('app_not_properly_removed', app=app))
```

Comment:
We don't do `_assert_system_is_sane_for_app(manifest, "post")` on remove operation? Dunno if that would be a good idea.

Reply:
Dunno either ... more generally I wish we could just find a way to reuse the regular `app_remove` instead of duplicating this code which generates inconsistencies when we make it evolve and forget to propagate stuff on both parts (not to mention the similar piece of code in `backup_restore`)